index.html

```diff
@@ -811,6 +814,94 @@ <h3>
           </p>
         </section>
       </section>
+      <section>
+        <h3>
+          `id` member
+        </h3>
+        <p>
+          The [=manifest's=] <code><dfn data-export="" data-dfn-for=
+          "manifest" id="identity">id</dfn></code> member is a <a>string</a> that
+          represents the <dfn>identity</dfn> for the application. 
+          The |identity| takes the form of a URL, which is same origin as the
+          start URL.
+        </p>
+        <p>The |identity| is used by user agents to uniquely identify the
+        application universally. When the user agent sees a manifest with an
+        |identity| that does not correspond to an already-installed application,
+        it SHOULD treat that manifest as a description of a distinct
+        application, even if it is served from the same URL as that of another
+        application. When the user agent sees a manifest with an |identity|
+        matching that of an already-installed application, it SHOULD be used as
+        a signal that this manifest is a replacement for the already-installed
+        application's manifest, and not a distinct application, even if it is
+        served from a different URL than the one seen previously.
+        </p>
+        <p>
+        The |identity| is considered matching when the processed |manifest|["id"] [=url/equal|equals=].
+        </p>
+        <p class="note">
+          The |identity| can be used by a service that collects lists of web
+          applications to uniquely identify applications.
+        </p>
+        <p>
+          To <dfn>process the `id` member</dfn>, given [=object=]
+          |json:JSON|, [=ordered map=] |manifest:ordered map|:
+        </p>
+        <ol class="algorithm">
+          <li>Set |manifest|["id"] to |manifest|["start_url"].
+          </li>
+          <li>If the type of |json|["id"] is not [=string=], return.
+          </li>
+          <li>Let |id:URL| be the result of [=URL Parser|parsing=] |json|["id"]
+          with |manifest|["start_ur"]'s origin as the base URL.
+          </li>
+          <li>If |id| is failure, return.
+          </li>
+          <li>If |id| is not same origin as |manifest|["start_url"], return.
+          </li>
+          <li>Otherwise, set |manifest|["id"] to |id|.</li>
+        </ol>
+        <aside class="example">
+          Below table shows some example cases of the |identity| processing algorithm.
+          <table class="data">
+            <tr>
+              <th>|json|["id"]</th>
+              <th>|manifest|["start_url"]</th>
+              <th>|manifest|["id"]</th>
+            </tr>
+            <tr>
+              <td><i>undefined</i></td>
+              <td>"https://example.com/resources/start"</td>
```

Comment:
Could we get some examples with subdomains, like used on GitHub Pages, e.g. w3c.github.io

Reply:
Also, maybe show examples of weirded id's like with spaces in them

Reply:
Yep, this table is fantastic for illustrating cases, but more cases would be better! I see on the [TAG review](https://github.com/w3ctag/design-reviews/issues/668) they asked about "What constitutes a valid ID?" -- I don't think this needs to be explicitly addressed in normative text, since the fact that it runs through the URL Parser is sufficient to specify that. But some examples would be helpful.

Maybe some Unicode characters. Some percent-encoded sequences to show that this is actually parsed as a URL (e.g. `%6D` should generate an id of ...`m`).

Also I would like to add an example with an input of "`/foo`" in addition to "`foo`", which should generate the same result. (As I have been saying in internal discussions, even though the leading slash is optional, I would very much like us to recommend using it in all examples, since having a URL without a leading slash be parsed relative to the origin is quite confunsing behaviour IMO and differs from how we parse all other URLs in the manifest.)

(Arguably, I would still like to see us parse the ID relative to the manifest URL for consistency with other URLs, and thus soft-mandate the leading slash, but that decision seems to have been made long ago and I'm not sure what the decision was based on.)

Reply:
For completeness, I guess we don't mind if the computed ID is not in scope as it's not actually involved in any navigations.  We might want to make a note of that.

Reply:
Added a note for the scope.
Added ASCII character and special character examples.